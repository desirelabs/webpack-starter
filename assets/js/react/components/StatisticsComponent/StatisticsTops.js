// @flow
import React from 'react'
import axios from 'axios'
import _ from 'lodash'
import { Row, Col } from 'react-bootstrap'
import Colors from '../../styles/colorScheme'

type Props = {
  program: string
}

type State = {
  today: Object,
  yesterday: Object
}

class StatisticsTops extends React.Component<Props, State> {
  constructor () {
    super()
    this.state = {
      today: {},
      yesterday: {}
    }
  }

  getData () {
    axios.get(window.Routing.generate(
      'program_stats_type',
      {
        type: 'visits',
        id: this.props.program
      }
    )).then(response => {
      this.setState({
        today: _.mapValues(_.invert(response.data.today)),
        yesterday: _.mapValues(_.invert(response.data.yesterday))
      })
    })
  }

  componentWillMount () {
    this.getData()
  }

  render () {
    return (
      <Row>
        <Col xs={12} md={6}>
          <h5 style={title}>Les plus visités</h5>
          <ul style={list}>
            {Object.values(this.state.today).map((label: any, index: any) => (
              index < 5 &&
                <li key={index}>
                  {(Object.values(this.state.today).indexOf(label) > Object.values(this.state.yesterday).indexOf(label)) && (
                    <i className='fa fa-arrow-right long-up' style={longUp} aria-hidden='true'>&nbsp;</i>
                  )}
                  {(Object.values(this.state.today).indexOf(label) < Object.values(this.state.yesterday).indexOf(label)) && (
                    <i className='fa fa-arrow-right long-down' style={longDown} aria-hidden='true'>&nbsp;</i>
                  )}
                  {(Object.values(this.state.today).indexOf(label) === Object.values(this.state.yesterday).indexOf(label)) && (
                    <i className='fa fa-arrow-right long-straight' style={longStraight} aria-hidden='true'>&nbsp;</i>
                  )}
                  {label}
                </li>
            ))}
          </ul>
        </Col>
        <Col xs={12} md={6}>
          <h5 style={title}>Les moins visités</h5>
          <ul style={list}>
            {Object.values(this.state.today).slice(Math.max(Object.values(this.state.today).length - 5, 1)).map((label: any, index: any) =>
              <li key={index}>
                {(Object.values(this.state.today).indexOf(label) > Object.values(this.state.yesterday).indexOf(label)) && (
                  <i className='fa fa-arrow-right long-up' style={longUp} aria-hidden='true'>&nbsp;</i>
                )}
                {(Object.values(this.state.today).indexOf(label) < Object.values(this.state.yesterday).indexOf(label)) && (
                  <i className='fa fa-arrow-right long-down' style={longDown} aria-hidden='true'>&nbsp;</i>
                )}
                {(Object.values(this.state.today).indexOf(label) === Object.values(this.state.yesterday).indexOf(label)) && (
                  <i className='fa fa-arrow-right long-straight' style={longStraight} aria-hidden='true'>&nbsp;</i>
                )}
                {label}
              </li>
            )}
          </ul>
        </Col>
      </Row>
    )
  }
}

const list = {
  paddingLeft: '5px',
  listStyleType: 'none'
}

const longUp = {
  transform: 'rotate(-45deg)',
  color: Colors.green
}

const longDown = {
  transform: 'rotate(45deg)',
  color: Colors.red
}

const longStraight = {
  color: Colors.orange
}

const title = {
  fontSize: '1.75rem',
  marginBottom: '1rem',
  paddingBottom: '1rem',
  borderBottom: '1px solid #d0d0d0'
}

export default StatisticsTops
