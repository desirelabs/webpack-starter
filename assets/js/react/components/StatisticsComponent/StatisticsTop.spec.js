import React from 'react'
import { configure, mount, shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'

import StatisticsTops from './StatisticsTops'

configure({adapter: new Adapter()})

describe('Testing StatisticsTops', () => {
  // Component should display
  test('Should display', () => {
    global.Routing = { generate: function () { return true } }
    const wrapper = renderer.create(<StatisticsTops />).toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})
