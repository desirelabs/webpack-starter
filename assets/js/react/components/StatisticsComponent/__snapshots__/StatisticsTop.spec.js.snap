// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing StatisticsTops Should display 1`] = `
<div
  className="row"
>
  <div
    className="col-md-6 col-xs-12"
  >
    <h5
      style={
        Object {
          "borderBottom": "1px solid #d0d0d0",
          "fontSize": "1.75rem",
          "marginBottom": "1rem",
          "paddingBottom": "1rem",
        }
      }
    >
      Les plus visités
    </h5>
    <ul
      style={
        Object {
          "listStyleType": "none",
          "paddingLeft": "5px",
        }
      }
    />
  </div>
  <div
    className="col-md-6 col-xs-12"
  >
    <h5
      style={
        Object {
          "borderBottom": "1px solid #d0d0d0",
          "fontSize": "1.75rem",
          "marginBottom": "1rem",
          "paddingBottom": "1rem",
        }
      }
    >
      Les moins visités
    </h5>
    <ul
      style={
        Object {
          "listStyleType": "none",
          "paddingLeft": "5px",
        }
      }
    />
  </div>
</div>
`;
